### PR TITLE
feat: web3 profile plugin

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Closes [#282](https://mask.atlassian.net/browse/MF-282)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 1]
```diff
@@ -44,7 +44,9 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
 
     const { t } = useI18N()
     const [hidden, setHidden] = useState(true)
+    const [addressList, setAddressList] = useState<AddressName[]>()
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx]: Missing default value.

File: packages/plugin-infra/src/types.ts [hunk 2]
```diff
@@ -69,7 +76,11 @@ export namespace Plugin.Shared {
         personaSign(payload: PersonaSignRequest): Promise<PersonaSignResult>
         /** Sign a message with wallet */
         walletSign(message: string, address: string): Promise<string>
+        /** Sign a message silently */
+        silentSign(signer: ECKeyIdentifier, message: string): Promise<PersonaSignResult>
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts]: This looks dangerous. I don't want to expose this ability to all plugins.

File: packages/plugins/Web3Profile/src/SNSAdaptor/types.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            amount: string
+        }[]
+        start_date?: string
+        end_date?: string
+        country?: string
+        city?: string
+    }
+}
+
+export interface personaKV {
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/types.ts]: Those interfaces should be upper cased.

File: packages/plugins/Web3Profile/src/SNSAdaptor/components/WalletSwitch.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    <Typography className={classes.accountName}>{getWalletName()}</Typography>
+                </div>
+                <div className={classes.infoRow}>
+                    <Typography className={classes.address} variant="body2" title={address?.address}>
+                        <FormattedAddress address={address?.address} size={4} formatter={formatAddress} />
+                    </Typography>
+
+                    <Link
+                        className={classes.link}
+                        href="www.baidu.com"
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/components/WalletSwitch.tsx]: ?

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/plugin-infra/src/types.ts (hunk 2 above)
Comment (Randolph314) [reply packages/plugin-infra/src/types.ts]: But I need this ability in web3 profile plugin

Comment (Jack-Works) [reply packages/plugin-infra/src/types.ts]: Rename it to `priviliged_silentSign` and only allow `plugin ID = web3 profile` plugin to use the implementation provided.

```js
priviliged_silentSign: async () => {
    if (pluginID !== PluginID.Web3Profile) throw new TypeError(...)
}
```

File: packages/mask/src/social-network/ui.ts [hunk 5]
```diff
@@ -120,6 +121,12 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
                 MaskMessages.events.currentPersonaIdentifier.on,
                 signal,
             )
+            const allPersonaSub = createSubscriptionFromAsync(
+                () => Services.Identity.queryOwnedPersonaInformation(true),
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts:127]: Do you sure you want to include the unintialized personas?

File: packages/mask/src/social-network/ui.ts [hunk 6]
```diff
@@ -137,9 +144,12 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
                 },
                 personaSign: Services.Identity.signWithPersona,
                 walletSign: Services.Ethereum.personalSign,
+                silentSign: Services.Identity.generateSignResult,
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts]: check plugin id here, if it is not web3 profile, throw an error

File: packages/mask/src/extension/dashboard/index.tsx [hunk 7]
```diff
@@ -48,6 +48,8 @@ startPluginDashboard(
             personaSign: Services.Identity.signWithPersona,
             walletSign: Services.Ethereum.personalSign,
             currentPersona: currentPersonaSub,
+            silentSign: Services.Identity.generateSignResult,
```
Comment (Jack-Works) [inline packages/mask/src/extension/dashboard/index.tsx:51]: Do you really need this silentSign everwhere? if not, please define it in the SNSContext or WorkerContext

File: packages/mask/src/plugins/NextID/icons/WalletIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+;<svg width="17" height="16" viewBox="0 0 17 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Jack-Works) [inline packages/mask/src/plugins/NextID/icons/WalletIcon.tsx:1]: ? How you're going to use this file?

File: packages/plugin-infra/src/types.ts (hunk 2 above)
Comment (Randolph314) [reply packages/plugin-infra/src/types.ts:80]: can I make it like this:
```Javascript
priviliged_silentSign: pluginID === PluginId.Web3Profile ? Services.Identity.generateSignResult
 : null,
```

File: packages/mask/src/social-network/ui.ts (hunk 5 above)
Comment (Randolph314) [reply packages/mask/src/social-network/ui.ts:127]: I need nickname and avatar url of current persona but currentPersona from ``` Services.Settings.getCurrentPersonaIdentifier``` doesn't include these

Comment (Jack-Works) [reply packages/mask/src/social-network/ui.ts:127]: I mean, `Services.Identity.queryOwnedPersonaInformation(true)` and `Services.Identity.queryOwnedPersonaInformation(false)` are different.

File: packages/mask/src/plugins/NextID/icons/WalletIcon.tsx (hunk 8 above)
Comment (Jack-Works) [reply packages/mask/src/plugins/NextID/icons/WalletIcon.tsx:1]: This is not resolved. How you're going to use this file? It has no export nor side effects.

File: packages/plugins/Web3Profile/src/SNSAdaptor/types.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    address: string
+    platform?: string
+    collections?: CollectionTypes[]
+}
+
+export interface accountType extends BindingProof {
+    walletList: WalletsCollection
+}
+
+export interface personaInfo {
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/types.ts]: Uppercase.

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 10]
```diff
@@ -55,20 +57,35 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
...
+    const { value: personaProof } = usePersonaBoundPlatform(
+        currentConnectedPersona?.identifier?.publicKeyAsHex || ZERO_ADDRESS,
+    )
+    const wallets = personaProof?.proofs?.filter((proof) => proof?.platform === NextIDPlatform.Ethereum)
 
-    const currentAccountNotConnectPersona =
-        currentIdentity.identifier === identity.identifier &&
-        personaList.findIndex((persona) => persona?.persona === currentConnectedPersona?.identifier.publicKeyAsHex) ===
-            -1
+    const isOwn = currentIdentity?.identifier === identity?.identifier
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx:65]: What if both is undefined?

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 11]
```diff
@@ -55,20 +57,35 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
...
+                return {
+                    type: AddressNameType.ADDRESS,
+                    label: proof?.identity,
+                    resolvedAddress: proof?.identity,
+                }
+            })
+            setAddressList([...addressNames, ...addresses])
+        }
+    }, [addressNames, wallets?.length, isOwn])
+    console.log({ addressList, currentConnectedPersona, personaList, platform, identity })
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx:80]: Remove log.

File: packages/plugin-infra/src/types.ts [hunk 12]
```diff
@@ -70,6 +77,8 @@ export namespace Plugin.Shared {
         /** Sign a message with wallet */
         walletSign(message: string, address: string): Promise<string>
         currentPersona: Subscription<PersonaIdentifier | undefined>
+        allPersona?: Subscription<PersonaInformation[]>
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts:81]: allPersonas

File: packages/mask/src/extension/dashboard/index.tsx (hunk 7 above)
Comment (Jack-Works) [reply packages/mask/src/extension/dashboard/index.tsx:51]: You did not do the permission check. Please only allow this on the plugin you want to.

File: packages/mask/src/extension/popups/render.tsx [hunk 13]
```diff
@@ -27,6 +27,8 @@ startPluginDashboard(
             personaSign: Services.Identity.signWithPersona,
             walletSign: Services.Ethereum.personalSign,
             currentPersona: currentPersonaSub,
+            silentSign: Services.Identity.generateSignResult,
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/render.tsx:30]: You did not do the permission check. Please only allow this on the plugin you want to.

File: packages/mask/src/plugins/NextID/components/NextIdPage.tsx [hunk 14]
```diff
@@ -109,18 +155,10 @@ export function NextIdPage({ personaList }: NextIdPageProps) {
...
-                {personaActionButton}
-            </Stack>
-        )
-    }
-
     if (loading || loadingPersona || loadingVerifyInfo) {
         return (
             <>
-                {Array.from({ length: 2 })
+                {Array.from({ length: 1 })
```
Comment (Jack-Works) [inline packages/mask/src/plugins/NextID/components/NextIdPage.tsx:169]: Then you don't need `Array.from`.

File: packages/mask/src/social-network/ui.ts [hunk 15]
```diff
@@ -137,9 +145,16 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
                 },
                 personaSign: Services.Identity.signWithPersona,
                 walletSign: Services.Ethereum.personalSign,
+                privileged_silentSign: () => {
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts:149]: It should not be a higher ordered function. Don't return a new function but call it directly.

File: packages/plugins/RSS3/src/types.ts [hunk 16]
```diff
@@ -48,3 +50,23 @@ export enum AssetType {
     GitcoinDonation = 'Gitcoin-Donation',
     POAP = 'POAP',
 }
+
+export interface patch {
```
Comment (Jack-Works) [inline packages/plugins/RSS3/src/types.ts:54]: Those types should be uppercased.

File: packages/plugins/Web3Profile/src/SNSAdaptor/components/ImageList.tsx [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+import { useI18N } from '../../locales'
+import { ImageIcon } from './ImageIcon'
+import { InjectedDialog } from '@masknet/shared'
+import type { PersonaInformation, NextIDStoragePayload } from '@masknet/shared-base'
+import type { CollectionTypes } from '../types'
+import { context } from '../context'
+import { getKvPayload, setKvPatchData } from '../hooks/useKV'
+
+const useStyles = makeStyles()((theme) => {
+    console.log({ theme })
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/components/ImageList.tsx:14]: Remove log.

File: packages/plugins/Web3Profile/src/SNSAdaptor/api.ts [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import urlcat from 'urlcat'
+import { Response, AssetType } from './types'
+
+async function fetchJSON<T = unknown>(url: string): Promise<T> {
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/SNSAdaptor/api.ts:4]:
```suggestion
async function fetchJSON(url: string): Promise<unknown> {
```

Don't cast, but check it's actual structure.

File: packages/plugins/Web3Profile/src/Worker/index.ts [hunk 19]
```diff
@@ -0,0 +1,9 @@
+import type { Plugin } from '@masknet/plugin-infra'
+import { base } from '../base'
+
+const worker: Plugin.Worker.Definition = {
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/src/Worker/index.ts:4]: Looks like you're not using it. Plz remove the worker entry.

File: packages/plugins/Web3Profile/README.md [hunk 20]
```diff
@@ -0,0 +1,7 @@
+# Example plugin
```
Comment (Jack-Works) [inline packages/plugins/Web3Profile/README.md:1]: Write it or remove it.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

**Pull request closed without merging**